src/configuration/remapper.ts

```diff
@@ -76,29 +79,173 @@ export class Remapper implements IRemapper {
+          // waits for the timeout to finish, this will resend the 'cc<BufferedKeys>'
+          // keys without allowing a potential remap on first key, which makes the
+          // first 'c' be handled as a 'ChangeOperator' and the second 'c' which has
+          // potential remaps (the 'ccc' remap) is buffered and the timeout started
+          // but then the '<BufferedKeys>' key comes straight away that clears the
+          // timeout without waiting again, and makes the second 'c' be handled normally
+          // as another 'ChangeOperator'.
+          //
+          // Example 2: if 'iiii' is mapped in normal and 'ii' is mapped in insert
+          // mode, and the user presses 'iii' in normal mode and waits for the timeout
+          // to finish, this will resend the 'iii<BufferedKeys>' keys without allowing
+          // a potential remap on first key, which makes the first 'i' be handled as
+          // an 'CommandInsertAtCursor' and goes to insert mode, next the second 'i'
+          // is buffered, then the third 'i' finds the insert mode remapping of 'ii'
+          // and handles that remap, after the remapping being handled the '<BufferedKeys>'
+          // key comes that clears the timeout and since the commandList will be empty
+          // we return true as we finished handling this sequence of keys.
+
+          keys = vimState.recordedState.commandList; // includes the '<BufferedKeys>' key
+
+          this._logger.debug(
+            `${this._configKey}. timeout finished, handling timed out buffer keys without allowing a new timeout.`
+          );
+        }
+        this._logger.debug(
+          `${this._configKey}. potential remap broken. resending keys without allowing a potential remap on first key. keys=${keys}`
+        );
+        this._hasPotentialRemap = false;
+        this._allowPotentialRemapOnFirstKey = false;
+        vimState.recordedState.resetCommandList();
+        await modeHandler.handleMultipleKeyEvents(keys);
+        return true;
+      }
+    }
+
+    if (isPotentialRemap && allowBufferingKeys && this._allowPotentialRemapOnFirstKey) {
+      if (remapping) {
+        // There are other potential remaps (ambiguous remaps), wait for other
+        // key or for the timeout to finish.
+        this._hasAmbiguousRemap = true;
+
+        this._logger.debug(
+          `${this._configKey}. ambiguous match found. before=${remapping.before}. after=${remapping.after}. command=${remapping.commands}. waiting for other key or timeout to finish.`
+        );
+      } else {
+        this._hasPotentialRemap = true;
+        this._logger.debug(
+          `${this._configKey}. potential remap found. waiting for other key or timeout to finish.`
+        );
+      }
+      vimState.recordedState.bufferedKeys = keys.slice(0);
+      vimState.recordedState.bufferedKeysTimeoutObj = setTimeout(() => {
+        modeHandler.handleKeyEvent('<BufferedKeys>');
+      }, configuration.timeout);
```

Comment:
I chose this key name but it can definitely be changed if someone thinks it is not obvious enough what it is about. Maybe it should have been `'<Timeout>'` or `'<TimeoutEnd>'`.

Reply:
I ended up changing it to `'<TimeoutFinished>'` since it better denotes what this key represents. I also added a SpecialKeys enum with this key and the extension enable and disable keys to make it easier to refactor if later we decide this isn't the right name for it.